python/ray/serve/_private/proxy_router.py

```diff
@@ -126,6 +143,29 @@ def match_route(
 
         return None
 
+    def ready_for_traffic(self, is_head: bool) -> Tuple[bool, str]:
+        """Whether the proxy router is ready to serve traffic.
+
+        The first return value will be false if any of the following hold:
+        - The route table has not been populated yet
+        - The route table has been populated, but none of the handles
+          have received running replicas yet AND it lives on a worker node.
+
+        Otherwise, the first return value will be true.
+        """
+
+        if not self._route_table_populated:
+            return False, NO_ROUTES_MESSAGE
+
+        if is_head:
+            return True, ""
```

Comment:
why is the head node an exceptional case?

Reply:
added comment